crates/proof-of-sql/src/sql/logical_plans/plan.rs

```diff
@@ -0,0 +1,128 @@
+use super::Expr;
+use crate::base::database::{ColumnField, TableRef};
+use alloc::{boxed::Box, vec::Vec};
+use serde::{Deserialize, Serialize};
+
+/// Enum of logical plans that are either provable or supported in postprocessing
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+pub enum LogicalPlan {
+    /// Empty
+    Empty,
+    /// Table scan
+    TableScan(TableRef),
```

Comment:
Give these variants actual `structs`.
Although there are not traits right now, I would rather design this so that something like `enum_dispatch` could easily be used if needed.
```suggestion
    /// Empty
    Empty(Empty),
    /// Table scan
    TableScan(TableScan),
```